Problems/15649/ans_15649_JHP.py

```diff
@@ -0,0 +1,70 @@
+# https://www.acmicpc.net/problem/15649
+
+# ---------------------------------------------------------------------
+# 알고리즘 - 백트래킹(Back-Tracking)
+# 정의 : 해를 찾는 도중 해가 아니어서 막히면, 되돌아가서 다시 해를 찾는 기법
+# 참고 자료 : https://chanhuiseok.github.io/posts/algo-23/
+# (1) 불필요한 부분을 쳐내고, 반복문의 횟수를 줄일 수 있어 효율적 
+# (2) N!의 경우의 수를 가진 뭄제에서는 최악의 경우 지수함수 시간을 필요함.
+# (3) 모든 경우의 수를 탐색하는 과정에서 조건문 등을 걸어 불필요한 부분 제거 
+# 즉, 얼마만큼 가지치기를 잘하느냐에 따라 효율성이 결정
+
+# 백트래킹 접근 방식 : 
+# (1) DFS 수행 - 깊이 우선탐색인 DFS를 수행하여 노드 탐색
+# (2) 유망한 노드 검토 - 방문한 노드를 포함해서 유망한 노드이면 서브트리로 이동, 아니면 백트래킹 수행
+# (3) 서브트리 이동 - 방문한 노드의 하위 노드로 이동하여 다시 재귀를 통해 DFS 수행
+# (4) 백트래킹 수행 -방문한 노드를 가지치기하고 사우이 노드로 백트래킹한 후 DFS를 다시 수행
+
+# ---------------------------------------------------------------------
+# 문제 : 자연수 N과 M이 주어졌을 때 1부터 N까지 자연수 중에서 중복 없이 M개를 고른 수열을 모두 구하기
+# 입력 : N, M (M <= N)
+# 출력 : 한 줄에 문제의 조건을 만족하는 수열 사전 순서대로 출력 (중복 수열 출력 금지)
+
+def find_num(idx,N,M):
+    global check
+    global result
+    
+    if idx == M:  # index가 M개가 되는 순간 수열 완료
+        for i in range(M):
+            print(result[i], end=' ')
+        print()
+        return
+    else :
+        for i in range(1,N+1):
```

Comment:
이 구문이 참 보기싫은데, 마땅한 방법이 없네.
```suggestion
        for i in range(N):
        	i += 1
```
나라면 이렇게 쓸 것 같은데, 이것도 딱히 readability가 좋은 코드는 아닌거같음.

Reply:
이게 시작이 1부터 올라 가는 변수라서..정말 마땅히 방법이 없더라고요...